internal/services/iothub/iothub_resource.go

```diff
@@ -1288,6 +1309,59 @@ func flattenIPFilterRules(in *[]devices.IPFilterRule) []interface{} {
 	return rules
 }
 
+func expandIotHubIdentity(input []interface{}) (*devices.ArmIdentity, error) {
+	config, err := iothubIdentity{}.Expand(input)
+	if err != nil {
+		return nil, err
+	}
+
+	var identityIds map[string]*devices.ArmUserIdentity
+	if len(config.UserAssignedIdentityIds) != 0 {
+		identityIds = map[string]*devices.ArmUserIdentity{}
+		for _, id := range config.UserAssignedIdentityIds {
+			identityIds[id] = &devices.ArmUserIdentity{}
+		}
+	}
```

Comment:
this'll ensure these are always removed:

```suggestion
	identityIds = make(map[string]*devices.ArmUserIdentity, 0)
	for _, id := range config.UserAssignedIdentityIds {
		identityIds[id] = &devices.ArmUserIdentity{}
	}
```

Reply:
Have updated the code, but I didn't fully understand the comment, is it related to the garbage collection mechanism?

And I found that identityIds has to be nil for a SystemAssigned identity. An error will be thrown if an empty map is given. So I've updated the code to first create the `ArmUserIdentity` without `identityIds` and set it only when provided. Please check if it looks good.